docs/source/architecture/data_services_architecture.rst

```diff
@@ -1,4 +1,110 @@
+Motech Data Services Architecture
+=================================
 
-There will be text here.
+MOTECH Data Services (MDS) is responsible for creation of Data Models for MOTECH Platform either using Developer Defined Entities (DDE) or End User Defined Entities (EUDE). DDEs use custom annotated POJO mark them as MDS entities. The EUDEs can be created as well as configured from UI along with their Data Access Layer. Centre of Data Access Layer is Generic Data Access Pattern Class which defines generic persistence related methods.
+
+         .. image:: mds_architecture.png
+            :scale: 100 %
+	  		:alt: Motech Data Services Architecture Diagram
+            :align: center
+
+
+        Section following describes motech data services architecture in figure.
+
+MDS UI Layer
+============
+
+#.Schema Editor: It is responsible for creating new EUDEs and also to edit already existing entities (Adding new fields and their lookups).
+#.Data Browser: It is responsible for browsing through entities.
+#.Settings: It is responsible for setting data retention policies .i.e. managing trash and history data. It also provides an option to export and import schema using csv as input or output.
+
+MDS Web
+=======
+
+It acts as controller between MDS UI Layer and MDS Core Layer to view, create and update EUDE and it’s infrastructure classes .i.e services and repositories and to view and update DDE and it’s infrastructure classes .i.e services and repositories. It use pre defined services of MDS.
+
+MDS Core Layer 
+==============
+
+#. MDS Initializer: It is responsible for building all entites in MDS database. It use MDS Constructor for building entites. It’s using TransactionalCallBackWithout Result of Spring framework for configuring Transaction. It includes MDSWeavingHook which is responsible for hooking into OSGi class loading process to dynamically add Jdo Imports and replacing DDE’s with thier bytecodes (extended version) generated by MDS.
+
+Added Jdo Imports are as follows:-
+javax.jdo,javax.jdo.identity,javax.jdo.spi,org.joda.time,org.apache.commons.lang,org.springframework.transaction.support,org.motechproject.mds.filter,org.motechproject.mds.query,org.motechproject.mds.util,org.motechproject.commons.date.util,org.datanucleus.enhancer,org.datanucleus,org.datanucleus.state,org.datanucleus.asm,org.datanucleus.exceptions,org.datanucleus.metadata,org.datanucleus.plugin,org.datanucleus.util,org.datanucleus.identity
+
+
+#. MDS Constructor: It provides methods to create entities, their service implementations and corresponding interfaces.
+
+#. Entity Builder: It is use to build classes of a given entities.
+
+#. Infrastructure Builder: It is responsible for building infrastructure for a given entity .i.e repository, interface and service classes. These classes are created only if not available in the classpath using  javassist to construct them.
+
+#. Metadata Builder: It is responsible for generating JDO metadata for entity.
```

Comment:
I think we can expand a bit more on this - for example that it uses both the schema and the class definition in case of the DDE for coming up with the result schema.